### PR TITLE
[isoltest] Error out in case of different result when enforcing via yul.

### LINKED ISSUE #10313: During "enforce yul" mode, different return value should be a hard error
In isoltest, when we are running in "enforce yul" mode, it should be a hard error if the yul version works but returns something else. Currently, this problem is silently skipped.

### PR DESCRIPTION
**Pull request by mijovic**
Fixes #10313

### CODE REVIEW

**Comment by mijovic** [comment]
Tests failing after rebase:
 - `semanticTests/dirty_memory_bytes_to_storgage_copy.sol` (#10677 fixes this one)
 - `semanticTests/cleanup/exp_cleanup_smaller_base.sol`
 - `semanticTests/cleanup/exp_cleanup_nonzero_base.sol`
 - `semanticTests/cleanup/exp_cleanup.sol`
 - `semanticTests/cleanup/exp_cleanup_direct.sol`
 - `semanticTests/revertStrings/called_contract_has_code.sol`
 - `semanticTests/revertStrings/unknown_sig_no_fallback.sol`
 - `semanticTests/revertStrings/ether_non_payable_function.sol`
 - `semanticTests/revertStrings/library_non_view_call.sol`
 - `semanticTests/various/code_access_padding.sol`
 - `semanticTests/exponentiation/literal_base.sol`
 - `semanticTests/exponentiation/small_exp.sol`
 - `semanticTests/libraries/library_address.sol`
 - `semanticTests/storage/packed_storage_overflow.sol`
 - `semanticTests/array/copying/storage_memory_nested_bytes.sol`

**Comment by chriseth** [comment]
Rebased.

**Comment by chriseth** [comment]
Failing tests:
unknown_sig_no_fallback, library_non_view_call, ether_non_payable_function, called_contract_has_code (#10765)
code_access_padding
library_address (Fixed)
packed_storage_overflow (#10762)

File: test/libsolidity/SemanticTest.cpp [hunk 1]
```diff
@@ -227,12 +227,15 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
...
 
-		if (success && !m_runWithYul && _compileViaYul)
+		if (!m_runWithYul && _compileViaYul)
 		{
-			m_compileViaYulCanBeSet = true;
-			AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) <<
+			m_compileViaYulCanBeSet = success;
+			string message = success ?
+				"Test can pass via Yul and marked with compileViaYul: false." :
+				"Test compiles viaYul, but it gives different result.";
```
Comment (axic) [inline test/libsolidity/SemanticTest.cpp]:
```suggestion
				"Test compiles via Yul, but it gives different test results.";
```

File: test/libsolidity/SemanticTest.cpp [hunk 2]
```diff
@@ -227,12 +227,15 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
...
 		}
 
-		if (success && !m_runWithYul && _compileViaYul)
+		if (!m_runWithYul && _compileViaYul)
 		{
-			m_compileViaYulCanBeSet = true;
-			AnsiColorized(_stream, _formatted, {BOLD, YELLOW}) <<
+			m_compileViaYulCanBeSet = success;
+			string message = success ?
+				"Test can pass via Yul and marked with compileViaYul: false." :
```
Comment (axic) [inline test/libsolidity/SemanticTest.cpp]:
```suggestion
				"Test can pass via Yul, but marked with \"compileViaYul: false.\"" :
```

**Review by axic — DISMISSED** [review]
Looks good to me, but obviously we're waiting for it to pass.

**Comment by mijovic** [comment]
After #10765 only remaining test will be `code_access_padding`

**Comment by axic** [comment]
Rebased after #10767.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
The `enforce_yul` test passed, so merging.

**Pull request merged**